src/Collector.cs

```diff
@@ -153,5 +143,26 @@ private static Socket Connect()
       _log.Info("Connected to collector.");
       return socket;
     }
+
+    private static bool IsSocketDisconnected (Socket socket)
+    {
+      // Is there any data available?
+      byte[] buffer = null;
+      while (socket.Poll(1, SelectMode.SelectRead))
+        {
+          // If no data is available then socket disconnected
+          if (socket.Available == 0)
+            return true;
+
+          // Clear socket data; we don't care what InstrumentApp sends
+          buffer = buffer ?? new byte[Math.Min(1024, socket.Available)];
+          do
+            {
+              socket.Receive(buffer);
+            }
+          while (socket.Available != 0);
```

Comment:
From UL's commit message:

```
If we never attempt to receive data from the socket then data that is available
in the receive buffer masks the fact that the socket has disconnected
```

I'm just going by that and observed behavior.  This code no longer exists, the behavior is different now.  Do we need to take this one all the way to satisfaction?  I'm fairly confident that the CLOSE_WAIT situation this was written to address is no longer possible.